### PR TITLE
Streamline OAuthRepository

### PR DESCRIPTION
**Pull request by juliepromact**
Coding conventions corrected in OAuthRepository

**Files changed** are :

1. **IOAuthRepository.cs** - Comments added.
2. **OAuthRepository.cs** - Used the concise form of object instantiation with implicit typing, added comments

### CODE REVIEW

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/OAuthRepository/OAuthRepository.cs [hunk 1]
```diff
@@ -40,24 +41,32 @@ private void Add(OAuth model)
...
+        /// <param name="email"></param>
+        /// <param name="clientId"></param>
+        /// <returns></returns>
+        private async Task<OAuth> OAuthClientChecking(string email, string clientId)
         {
             //checking whether with this app email is register or not if  not new OAuth will be created.
-            var oAuth = GetDetails(email, clientId);
-            var app =  await _appRepository.GetAppDetails(clientId);
-            if (oAuth == null && app!=null)
+            OAuth oAuth = GetDetails(email, clientId);
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/OAuthRepository/OAuthRepository.cs]: Why are you calling GetDetails without async support?

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/OAuthRepository/OAuthRepository.cs [hunk 2]
```diff
@@ -69,57 +78,77 @@ private OAuth GetDetails(string email,string clientId)
...
+        /// <param name="redirectUrl"></param>
+        /// <param name="refreshToken"></param>
+        /// <returns></returns>
         private async Task<OAuthApplication> GetAppDetailsFromClient(string redirectUrl, string refreshToken)
         {
             // Assigning Base Address with redirectUrl
-            var requestUrl = string.Format("?refreshToken={0}", refreshToken);
-            var response = await _httpClientRepository.GetAsync(redirectUrl, requestUrl);
-            var responseResult = response.Content.ReadAsStringAsync().Result;
+            string requestUrl = string.Format("?refreshToken={0}", refreshToken);
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/OAuthRepository/OAuthRepository.cs]: Just a test

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/OAuthRepository/OAuthRepository.cs (hunk 1 above)
Comment (juliepromact) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/OAuthRepository/OAuthRepository.cs]: GetDetails is a private method and it just gets data from database. Do we need to give async support for such methods too ?

Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/OAuthRepository/OAuthRepository.cs]: yes, if it connects with DB it needs to be async

Comment (juliepromact) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/OAuthRepository/OAuthRepository.cs]: Ok. Will change code accordingly.

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Services/CustomAttribute.cs [hunk 3]
```diff
@@ -16,7 +16,7 @@ public CustomAttribute(IOAuthRepository oAuthRepository)
         public override void OnActionExecuting(ActionExecutingContext filterContext)
         {
             var token = filterContext.HttpContext.Request.Headers["Authorization"].ToString();
-            var data = _oAuthRepository.GetDetailsClientByAccessToken(token);
+            var data = _oAuthRepository.GetDetailsClientByAccessToken(token).Result;
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Services/CustomAttribute.cs]: This should be awaited

**Pull request merged**